java/core/src/main/java/com/lancedb/lance/schema/ColumnAlteration.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.lancedb.lance.schema;
+
+import org.apache.arrow.vector.types.pojo.ArrowType;
+
+import java.util.Optional;
```

Comment:
It's better use the `Optional` in java-core module since the jdk `Optinal` is not seriable. And the spark connector need a seriable class.

Reply:
> It's better use the Optional in java-core module

What does this mean? Which `Optional` do you prefer?  I see, other core classes for example `Fragment`, `FragmentOperation`, `WriteParams`, `ReadOptions` also use `java.util.Optional`.

Reply:
OK, maybe the `Optional` in java should be named as `SeriableOptional` for spark to use.